edgelet/iotedged/src/lib.rs

```diff
@@ -253,21 +253,44 @@ impl Main {
             .context(ErrorKind::Initialize(InitializeErrorReason::Hsm))?;
         info!("Finished initializing hsm.");
 
-        // Detect if the settings were changed and if the device needs to be reconfigured
         let cache_subdir_path = Path::new(&settings.homedir()).join(EDGE_SETTINGS_SUBDIR);
-        check_settings_state(
-            cache_subdir_path.clone(),
-            EDGE_SETTINGS_STATE_FILENAME,
-            &settings,
-            &runtime,
-            &crypto,
-            &mut tokio_runtime,
-        )?;
-
         macro_rules! start_edgelet {
-            ($key_store:ident, $provisioning_result:ident, $root_key:ident, $runtime:ident) => {{
+            ($key_store:ident, $provisioning_result:ident, $root_key:ident) => {{
                 info!("Finished provisioning edge device.");
 
+                let runtime: M = init_runtime(
+                    settings.clone(),
+                    &mut tokio_runtime,
+                    $provisioning_result.clone(),
+                )?;
+
+                if $provisioning_result.reconfigure() != ReprovisioningStatus::DeviceDataNotUpdated {
```

Comment:
Are you sure this will only be true in the `dps_*_provision` cases (which is where this code originally belonged) and not in manual / external cases?

Reply:
Manual and external provisioning currently always return `ReprovisioningStatus::DeviceDataNotUpdated`. I suppose you're asking if we should also check here that provisioning mode is DPS. Seems to me that if removing all containers in response to a re-provisioning event is appropriate for DPS then it'd be appropriate for other modes as well in case we started returning anything other than `ReprovisioningStatus::DeviceDataNotUpdated` in the future. What do you think?

Reply:
>Seems to me that if removing all containers in response to a re-provisioning event is appropriate for DPS then it'd be appropriate for other modes as well in case we started returning anything other than `ReprovisioningStatus::DeviceDataNotUpdated` in the future for those modes. What do you think?

Yes, I would assume this as well, but I wanted to be sure that this assumption is valid. If you're sure then it's okay.

Maybe tweak the "If there was a DPS reprovision" comment below to account for other provisioning methods.

Reply:
Cool. Verified that manual/external don't use the other reprovision status variants. Updated the comment.